src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/DefaultRazorComponentSearchEngineTest.cs

```diff
@@ -0,0 +1,99 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using Microsoft.AspNetCore.Mvc.Razor.Extensions.Version2_X;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.Language.Components;
+using Microsoft.AspNetCore.Razor.Language.Intermediate;
+using Microsoft.AspNetCore.Razor.Test.Common;
+using Microsoft.CodeAnalysis.Razor.ProjectSystem;
+using Moq;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Test.Refactoring
+{
+    public class DefaultRazorComponentSearchEngineTest : LanguageServerTestBase
+    {
+        private static ProjectSnapshotManager _projectSnapshotManager = CreateProjectSnapshotManager();
+        private static readonly DefaultRazorComponentSearchEngine _searchEngine = new DefaultRazorComponentSearchEngine(_projectSnapshotManager);
+
+        [Theory]
+        [InlineData("First", "First.Components", "Component1")]
```

Comment:
We actually try to rarely use `InlineData` because even though it reduces code re-use it makes it near impossible to understand why a certain test is failing. For instance, imagine this test fails you now have to understand the context behind each of these inline data items. In essence these should each be their own test with their own description as to what they're verifying.

You try and account for this below with comments but if the test fails you wont see the comments/know which variant of the inline data failed. I'd suggest the same for that as well